src/mz/src/vault.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Result;
+use clap::Parser;
+use serde::{de::Visitor, Deserialize, Serialize};
+
+use crate::configuration::FronteggAPIToken;
+
+const INLINE_PREFIX: &str = "mzp_";
+
+/// A vault stores sensitive information and provides
+/// back a token than can be used for later retrieval.
+/// By default, credentials are stored in the OS specific
+/// keychain if it exists.
+#[derive(Parser, Debug)]
+#[clap(group = clap::ArgGroup::new("vault").multiple(false))]
+pub struct Vault {
+    /// Inline the credentials in the CLIs
+    /// configuration file.
+    #[clap(long, short, group = "vault")]
```

Comment:
#[clap(long, short, group = "vault")]
```suggestion
    #[clap(long, group = "vault")]
```

Reply:
Collision with `--interactive -i`

Reply:
good catch!